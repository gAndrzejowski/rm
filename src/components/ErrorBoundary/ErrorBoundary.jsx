import React from 'react';

class ErrorBoundary extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            error:null
        }
    }

    componentDidCatch(error, info) {
        this.setState({error: {
                error,
                info
            }});
    }
    render() {
        return this.state.error ? (<h1>Something went horribly wrong!</h1>) : this.props.children;
    }
}
export default ErrorBoundary;