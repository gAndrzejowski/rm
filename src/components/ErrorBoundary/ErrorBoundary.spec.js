import React from 'react';
import renderer from 'react-test-renderer';
import ErrorBoundary from './ErrorBoundary';

describe('Error boundary', () => {
  it('renders correctly', () => {
    const test = renderer.create(<ErrorBoundary><h1>:)</h1></ErrorBoundary>);
    expect(test.toJSON()).toMatchSnapshot();

    test.getInstance().componentDidCatch('a', 'b');
    expect(test.toJSON()).toMatchSnapshot();
  });
});
