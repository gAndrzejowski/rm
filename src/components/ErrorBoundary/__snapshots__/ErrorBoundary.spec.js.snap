// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error boundary renders correctly 1`] = `
<h1>
  :)
</h1>
`;

exports[`Error boundary renders correctly 2`] = `
<h1>
  Something went horribly wrong!
</h1>
`;
